.github/workflows/test-query-compiler-template.yml

```diff
@@ -100,4 +100,4 @@ jobs:
       - name: 'Run tests'
         env:
           PARTITION: ${{ matrix.partition }}
-        run: cargo nextest run --package query-engine-tests --partition hash:"$PARTITION"
+        run: cargo nextest run --package query-engine-tests --test-threads=8 --partition hash:"$PARTITION"
```

Comment:
I think we only have 4 cores: https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories.

Running 2 threads per core is probably okay given how much IO our tests are doing though.

Reply:
might be worth it to experiment with what number of test threads performs the best though

Reply:
No time for that now, but will consider extracting the number of test threads as a top level variable later.